app/components/windows/SourceFilters.vue.ts

```diff
@@ -87,19 +123,21 @@ export default class SourceFilters extends Vue {
   }
 
   removeFilter() {
-    this.sourceFiltersService.remove(this.sourceId, this.selectedFilterName);
-    this.filters = this.sourceFiltersService.getFilters(this.sourceId);
-    this.selectedFilterName = (this.filters[0] && this.filters[0].name) || null;
+    this.editorCommandsService.executeCommand(
+      'RemoveFilterCommand',
+      this.sourceId,
+      this.selectedFilterName,
+    );
   }
 
   toggleVisibility(filterName: string) {
     const sourceFilter = this.filters.find(filter => filter.name === filterName);
-    this.sourceFiltersService.setVisibility(
+    this.editorCommandsService.executeCommand(
+      'EnableFilterCommand',
```

Comment:
Yeah.  Well, it's not a toggle since it doesn't switch to the opposite of the state it's currently in (although it could be changed to do that).  A long form name would be something like `SetFilterEnabledStateCommand`.  I felt that was a little verbose but it's probably better than using something called `EnableFilterCommand` to disable a filter.

Reply:
isn't it setting it to an opposite state via `!sourceFilter.visible` unless im misreading something here

Reply:
Yeah I somewhat stupidly made the rollback command a toggle, even though the execute command isn't.  I should fix that.

Reply:
how does one turn visibility off in this case? the parent function still appears to assume this functions as a toggle

Reply:
The parent function was translating the toggle into a set, since the service expects a set, but the UI has it implemented as a toggle.  I just refactored this command to be a toggle instead to better match the UI rather than the service, since commands are meant to reflect user actions.